js/data/countries.js

```diff
@@ -0,0 +1,1529 @@
+    currency: 'TMT',
+    number: '934',
+  },
+  {
+    name: 'Turks and Caicos Islands (The)',
+    dataName: 'TURKS_AND_CAICOS_ISLANDS',
+    currency: 'USD',
+    number: '840',
+  },
+  {
+    name: 'Tuvalu',
+    dataName: 'TUVALU',
+    currency: 'AUD',
+    number: '036',
+  },
+  {
+    name: 'Uganda',
+    dataName: 'UGANDA',
+    currency: 'UGX',
+    number: '800',
+  },
+  {
+    name: 'Ukraine',
+    dataName: 'UKRAINE',
+    currency: 'UAH',
+    number: '980',
+  },
+  {
+    name: 'United Arab Emirates (The)',
+    dataName: 'UNITED_ARAB_EMIRATES',
+    currency: 'AED',
+    number: '784',
+  },
+  {
+    name: 'United Kingdom of Great Britain and Northern Ireland (The)',
+    dataName: 'UNITED_KINGDOM',
+    currency: 'GBP',
+    number: '826',
+  },
+  {
+    name: 'United States of America (The)',
+    dataName: 'UNITED_STATES',
+    currency: 'USD',
+    number: '840',
+  },
+  {
+    name: 'Uruguay',
+    dataName: 'URUGUAY',
+    currency: 'UYU',
+    number: '858',
+  },
+  {
+    name: 'Uzbekistan',
+    dataName: 'UZBEKISTAN',
+    currency: 'UZS',
+    number: '860',
+  },
+  {
+    name: 'Vanuatu',
+    dataName: 'VANUATU',
+    currency: 'VUV',
+    number: '548',
+  },
+  {
+    name: 'Venezuela (Bolivarian Republic Of)',
+    dataName: 'VENEZUELA',
+    currency: 'VEF',
+    number: '937',
+  },
+  {
+    name: 'VietNam',
+    dataName: 'VIETNAM',
+    currency: 'VND',
+    number: '704',
+  },
+  {
+    name: 'Virgin Islands (British)',
+    dataName: 'VIRGIN_ISLANDS_BRITISH',
+    currency: 'USD',
+    number: '840',
+  },
+  {
+    name: 'Virgin Islands (U.S.)',
+    dataName: 'VIRGIN_ISLANDS_US',
+    currency: 'USD',
+    number: '840',
+  },
+  {
+    name: 'Wallis and Futuna',
+    dataName: 'WALLIS_AND_FUTUNA',
+    currency: 'XPF',
+    number: '953',
+  },
+  {
+    name: 'Western Sahara',
+    dataName: 'WESTERN_SAHARA',
+    currency: 'MAD',
+    number: '504',
+  },
+  {
+    name: 'Yemen',
+    dataName: 'YEMEN',
+    currency: 'YER',
+    number: '886',
+  },
+  {
+    name: 'Zambia',
+    dataName: 'ZAMBIA',
+    currency: 'ZMW',
+    number: '967',
+  },
+  {
+    name: 'Zimbabwe',
+    dataName: 'ZIMBABWE',
+    currency: 'ZWL',
+    number: '932',
+  },
+];
+
+export default countries;
+
+let indexedCountries;
+
+function getIndexedCountries() {
+  if (indexedCountries) return indexedCountries;
+
+  indexedCountries = countries.reduce((indexedObj, country) => {
+    indexedObj[country.dataName] = _.omit(country, 'dataName');
+    return indexedObj;
+  }, {});
+
+  return indexedCountries;
+}
+
+export function getCountryByDataName(dataName) {
+  if (!dataName) {
+    throw new Error('Please provide a dataName.');
+  }
+
+  return getIndexedCountries()[dataName];
+}
+
+function getTranslatedCountries(lang, sort = true) {
+  if (!lang) {
+    throw new Error('Please provide the language the translated couuntries' +
+      ' should be returned in.');
+  }
+
+  let translated = [...countries];
```

Comment:
because sort, sort's the array in place and I don't want to modify the main list.